### ISSUE TITLE
Dowloading release assets

### ISSUE THREAD

**Issue by Bowbaq** [google/go-github#207]
I'm trying to use `go-github` to download assets from a release. Here's the code I have:
```go
// Based on https://developer.github.com/v3/repos/releases/#get-a-single-release-asset
func githubDownloadReleaseAsset(asset *github.ReleaseAsset, out io.Writer) error {
	gh := githubClient() // Initializes *github.Client

	req, err := gh.NewRequest("GET", *asset.URL, nil)
	req.Header.Set("Accept", "application/octet-stream")
	if err != nil {
		return err
	}

	_, err = gh.Do(req, out)
	return err
}
```
It seems the GitHub API redirects to a pre-signed URL for an S3 bucket, but I'm getting a `400` error, and I can't quite figure out why. Any help would be much appreciated.

**Comment by Bowbaq** [comment]
After some fiddling around, it seems the issue is that following the redirect to S3 using the oauth http client fails. Here's my workaround so far:
```go
func githubDownloadReleaseAsset(token string, asset *github.ReleaseAsset, out io.Writer) error {
	client := oauth2.NewClient(oauth2.NoContext, tokenSource{
		&oauth2.Token{AccessToken: token},
	})

	req, err := http.NewRequest("GET", *asset.URL, nil)
	if err != nil {
		return err
	}
	req.Header.Set("Accept", "application/octet-stream")

	res, err := client.Transport.RoundTrip(req)
	if err != nil {
		return err
	}

	s3URL, err := res.Location()
	if err != nil {
		return err
	}

	res, err = http.Get(s3URL.String())
	if err != nil {
		return err
	}
	defer res.Body.Close()

	_, err = io.Copy(out, res.Body)

	return err
}
```

**Comment by akshaychhajed** [comment]
#209 fixes this. Why didnt this issue autoclose?

**Comment by shurcooL** [comment]
@akshaychhajed Thanks for pointing that out.

Closing, this has been resolved in 25b0481acbe9f2123f2b39c3235b156cff6296cd.

> Why didnt this issue autoclose?

That PR was merged by rebasing on master and manually merging a new commit. That commit didn't say it resolves this issue in the commit message. The PR was closed manually, but this issue was forgotten.
